internal/cli/configuration.go

```diff
@@ -184,7 +198,40 @@ func ConfigurationCreate(cmd *cobra.Command, args []string) error {
 		return errors.Wrap(err, "error initializing cli")
 	}
 
-	err = client.CreateConfiguration(args[0], args[1:])
+	kvFromFiles, err := cmd.Flags().GetStringSlice("from-file")
+	if err != nil {
+		return errors.Wrap(err, "failed to read option --from-file")
+	}
+
+	// regular key/value assignments from the command line
+	kvDict := args[1:]
+
+	// extend with key/value assignments from files (--from-file option).
+	if len(kvFromFiles) > 0 {
+		for _, spec := range kvFromFiles {
+			pieces := strings.SplitN(spec, "=", 2)
+			var key string
+			var valuefile string
+			if len(pieces) < 2 {
+				// Argument is path only. Set key as last element of that path
+				valuefile = pieces[0]
+				_, key = path.Split(valuefile)
+			} else {
+				// Argument is key=path
+				key = pieces[0]
+				valuefile = pieces[1]
+			}
+			// Read file content ...
+			content, err := os.ReadFile(valuefile)
+			if err != nil {
+				return errors.Wrapf(err, "filesystem error")
+			}
+			// ... and add to command line
+			kvDict = append(kvDict, key, string(content))
+		}
+	}
+
+	err = client.CreateConfiguration(args[0], kvDict)
```

Comment:
It seems too much logic to me inside this func that should probably only be a "passthrough" to the `client.CreateConfiguration` that should handle this.

I would keep only the check key-filename logic (with the handling of the missing `=`), but adding one more arg to the `client.CreateConfiguration` func:

```
func (c *EpinioClient) CreateConfiguration(name string, dict []string, fileDict []string) error
```

The `fileDict` is an array of `key=filename`, and the underlying func will handle the reading of the files, maybe in a separate function.

Reply:
I am willing to place the file k/v logic into a separate function to make the cmd runner smaller.
However I would really like to keep that logic here in the CLI frontend.
  - It keeps front- and backend properly decoupled.
  - The backend receives only a list of key/val data as before and does not have to care where they came from, plain cli argument, or file. It does not change at all.

Reply:
Done with commit 03d2e28d4e39abaaf263ec52671a7e998712d6d3 (my proposal)